vhd/lib/vhd-util-coalesce.c

```diff
@@ -100,6 +100,10 @@ vhd_util_coalesce_block(vhd_context_t *vhd, vhd_context_t *parent,
 	if (err)
 		goto done;
 
+	err = posix_memalign((void *)&buf, 4096, vhd->header.block_size);
+	if (err)
+		return -err;
```

Comment:
Given the `goto done` immediately above, shouldn't this match ?

Reply:
yes, it should as the order of allocation of the block and bitmap has changed so we need to free the bitmap if we fail to memalign the block.